src/services/identity/SitesCacheService.ts

```diff
@@ -14,81 +18,232 @@ import type { GitHubRepositoryData, RepositoryData } from "@root/types/repoInfo"
+    // (In the same manner has the links for the first page do not prev and first links)
+    return firstPageRepos
+  }
+
+  // There are more pages to retrieve! Fetch all pages 2 to N in parallel, as was done before
+  const pageNums = _.range(2, links.last.pagenum + 1)
+
+  const pages2ToLast = await Promise.all(
+    pageNums.map((page) => fetchPageOfRepositories({ accessToken, page }))
   )
 
-  const allSites = await Promise.all(
-    paramsArr.map(async (params) => {
-      const {
-        data: respData,
-      }: {
-        data: GitHubRepositoryData[]
-      } = await genericGitHubAxiosInstance.get(
-        GITHUB_ORG_REPOS_ENDPOINT,
-        accessToken
-          ? {
-              headers: { Authorization: `token ${accessToken}` },
-              params,
-            }
-          : { params }
-      )
-      return respData
-        .map((gitHubRepoData) => {
-          const {
-            pushed_at: updatedAt,
-            permissions,
-            name,
-            private: isPrivate,
-          } = gitHubRepoData
-
-          return {
-            lastUpdated: updatedAt,
-            permissions,
-            repoName: name,
-            isPrivate,
-          } as RepositoryData
-        })
-        .filter((repoData) => {
-          if (!repoData || !repoData.permissions) {
-            return false
-          }
-          return (
-            repoData.permissions.push === true &&
-            !ISOMER_ADMIN_REPOS.includes(repoData.repoName)
-          )
-        })
+  return firstPageRepos.concat(...pages2ToLast.map((res) => res.repos))
+}
+
+async function getAllRepoDataSequentially() {
+  // This function exists because when we refresh the cache on interval, we want to optimize for load
+  // distribution in the process over time, rather than optimize for time to refresh
+
+  const allRepos = []
+  let page = 1
+
+  // We disable the eslint rule no-await-in-loop, because we specifically want sequence here to distribute load
+  /* eslint-disable no-await-in-loop */
+  do {
+    const { repos, links } = await fetchPageOfRepositories({
+      page,
+      getLinks: true,
     })
-  )
-  return _.flatten(allSites)
+
+    allRepos.push(...repos)
+
+    if (!links?.next) {
+      break
+    }
+
+    page = links.next.pagenum // Note that by right, we should follow the next link, rather than extract the page number from the link
+  } while (true) // eslint-disable-line no-constant-condition
+  /* eslint-enable no-await-in-loop */
+
+  return allRepos
 }
 
 export class SitesCacheService {
-  private repoDataCache: RepositoryData[]
+  private repoDataCache: CacheStore
 
   private refreshInterval: number
 
   constructor(refreshInterval: number) {
-    this.repoDataCache = []
-    this.refreshInterval = refreshInterval
-    this.renewCache()
+    this.repoDataCache = {} as CacheStore
+    this.refreshInterval = 90000
```

Comment:
should this be `this.refreshInterval = refreshInterval`? we can give a default to get this behaviour

Reply:
😱 Oh yes! That was not supposed to be committed! 😱

Reply:
Fixed in https://github.com/isomerpages/isomercms-backend/pull/1239/commits/c5aef04f88639ff759f8a3715611483547e2e70c
Good catch! 🙏